src/b2aiprep/process.py

```diff
@@ -172,18 +176,45 @@ def to_features(
         md5sum = md5(f.read()).hexdigest()
     audio = Audio.from_file(filename)
     audio = audio.to_16khz()
-    features = specgram(audio)
-    features_melfilterbank = melfilterbank(features, n_mels=n_mels)
+    features_specgram = specgram(audio)
+    features_melfilterbank = melfilterbank(features_specgram, n_mels=n_mels)
     features_mfcc = MFCC(features_melfilterbank, n_coeff=n_coeff, compute_deltas=compute_deltas)
     features_opensmile = extract_opensmile(audio, opensmile_feature_set, opensmile_feature_level)
+    
     features = {
-        "specgram": features,
+        "specgram": features_specgram,
         "melfilterbank": features_melfilterbank,
         "mfcc": features_mfcc,
         "opensmile": features_opensmile,
         "sample_rate": audio.sample_rate,
         "checksum": md5sum,
     }
-    outfile = outdir / f"sub-{subject}_task-{task}_md5-{md5sum}_features.pt"
+    
+    #outfile = outdir / f"sub-{subject}_task-{task}_md5-{md5sum}_features.pt"
+    #saving file with just UUID name for now. TO DO: change it back possibly
+    feat_file_name = Path(filename).stem
+    outfile = outdir / f"{feat_file_name}_features.pt"
+    
     torch.save(features, outfile)
-    return features, outfile
+    
+    if save_figures:
+        #save spectogram as figure
+        
+        #general log spectrogram for image
+        features_specgram_log = specgram(audio, log=True)
+        
+        fig, ax = plt.subplots(figsize = (20,10))
+        ax.matshow(features_specgram_log, origin='lower', aspect=8, norm=LogNorm(vmin=0, vmax=1))
```

Comment:
already getting log specgram.

```suggestion
        ax.matshow(features_specgram_log, origin='lower', aspect=8)
```